packages/smooth_ui_library/lib/dialogs/smooth_AlertDialog.dart

```diff
@@ -13,72 +13,99 @@ import 'package:smooth_ui_library/buttons/smooth_simple_button.dart';
 ///
 
 class SmoothAlertDialog extends StatefulWidget {
-  const SmoothAlertDialog(
+  final BuildContext context;
+  final String title;
+  final bool close;
+  final double height;
+  final Widget body;
+  final List<SmoothSimpleButton> actions;
+
+  SmoothAlertDialog(
       {Key key,
+      this.context,
       this.title,
       this.close = true,
       this.height,
       this.body,
       this.actions})
       : super(key: key);
 
+  @override
+  _SmoothAlertDialogState createState() => _SmoothAlertDialogState(
+        key: key,
+        context: context,
+        title: title,
+        close: close,
+        height: height,
+        body: body,
+        actions: actions,
+      );
+}
+
+class _SmoothAlertDialogState extends State<SmoothAlertDialog> {
+  final BuildContext context;
   final String title;
   final bool close;
   final double height;
   final Widget body;
   final List<SmoothSimpleButton> actions;
 
-  @override
-  _SmoothAlertDialogState createState() => _SmoothAlertDialogState();
-}
-
-class _SmoothAlertDialogState extends State<SmoothAlertDialog> {
+  _SmoothAlertDialogState(
```

Comment:
From what I understood of `StatefulWidget`s, you should keep the constants in the `StatefulWidget`, and leave only the variables in the `State<>` class.
Therefore, keep the `widget.xxx` notation.

Reply:
What do you mean, I don't think I can follow you here.

Reply:
This is fixed in the Master branch, you can just use the smooth_alert_dialog.dart from there :wink: 
What @monsieurtanuki meant is you don't need to declare the final variables in State as the ones from the Widget are accessible using `widget.attribute`

Reply:
I mean you're supposed not to have any parameters:
```dart
_SmoothAlertDialogState createState() => _SmoothAlertDialogState();
```
and to use whatever parameter fields
* from the `StatefulWidget`
* inside the `State<...>` class
* with the `widget.field` syntax

Reply:
I have never heard of it. But it is now installed thanks for the tip